lib/matplotlib/colors.py

```diff
@@ -674,14 +675,23 @@ def _create_lookup_table(N, data, gamma=1.0):
     return np.clip(lut, 0.0, 1.0)
 
 
-class Colormap:
```

Comment:
I'm not convinced we need this subclassing. Either define a minimal common protocol class, or create a union type for type annotations.

Reply:
We should definitely have an example of how people can make their own custom colormap classes that mix colors in interesting ways that matplotlib does not support natively. This can be done by subclassing `ColormapBase` or `MultivarColormap`, and this may have different behavior if the user then tries to visualize the colorbars [i.e. a multivariate form of `fig.colorbar(cm)`]. 
In this regard, there might be some additional flexibility if we include `ColormapBase` but this is somewhat hypothetical at this point.

As @story645 mentioned `ColormapBase` was introduced to simplify the use of `isinstance`.

Reply:
> We should definitely have an example of how people can make their own custom colormap classes

Let's get not ahead of ourselves. There may be some niche cases, but they are rare. I assume that 99% of users only need the currently available single-val `Colormap`. A large fraction of the remaining cases are addressed by the new colormap types here. While keeping possible further extensions in mind, we don't have to address them for now.


> As @story645 mentioned `ColormapBase` was introduced to simplify the use of `isinstance`.

Something feels odd here. What is the use of `isinstance` when the instance does not define behavior?

Reply:
I see what you mean. We can probably make some changes so that `isinstance` is no longer required.

Reply:
Regarding the original question, the `ColormapBase` class has been removed.